File: client-app/src/app/pages/post-page/model/post.ts
export class Post {
    id: number;
    author: string;
    date: string | null;
    displayDate: string;
    edited: boolean;
    text: string;
    isNew: boolean;
    fileDescDto: fileDescDto[];

    constructor() {
        this.text = '';
        this.edited = false;
        this.isNew = false;
        this.date = null;
        this.displayDate = '';
        this.author = '';
        this.id = -1;
        this.fileDescDto = []
    }
}

export class fileDescDto {
    name: string;
    size: string;

    constructor() {
        this.name = '';
        this.size = '0';
    }
}

File: WebService.Infrastructure/Entity/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebService.Infrastructure.Entity
{
    [Table("post")]
    public class Post
    {
        [Column("id")]
        [Key]
        public int Id { get; set; }

        [Column("author")]
        public string Author { get; set; }

        [Column("date")]
        public DateTime Date { get; set; }

        [Column("edited")]
        public bool Edited { get; set; }

        [Column("text")]
        public string Text { get; set; }

        public List<PostFile> PostFile { get; set; }

    }
}
